RereleaseTimer/index.js

```diff
@@ -1,203 +1,100 @@
-                                                };
-                                                PatchRelease(token, owner, app, release.id, patch)
-                                                .then((release) => {
-                                                    resolve()
-                                                })
-                                                .error(response => {
-                                                    context.error(response);
-                                                    reject(response);
-                                                });
-                                            }
-                                        });
-
-                                        resolve(true);
-                                    } else {
-                                        reject("Could not lookup destination for re-release.");
-                                    }
-                                });
-                            } else {
-                                context.log(`Nothing to perform.`);
-                                resolve(false);
-                            }
-                        });
-                    } else {
-                        context.log(`Latest release (${release.short_version}) has already been distributed to the destination.`);
-                        resolve(false);
-                    }
-                } else {
-                    context.log("No releases available in source.");
-                    resolve(false);
-                }
-            })
-            .error(response => {
-                reject(error);
-                context.error(response);
-            });
-        }));
+    for (rule of rules) {
+        ruleSet.push(getRulePromise(rule, context));
     }
 
     Promise.all(ruleSet)
-    .then(values => {
-        context.log("Finished processing!");
-        context.done();
-    });
+        .then(values => {
+            context.log("Finished processing!");
+            context.done();
+        }).catch((error) => {
+            context.log(error);
+            context.done();
+        });
 };
 
-function BuildUrl(endpoint, token, owner, app) {
-    const options = {
-        headers: { 'Accept': 'application/json', 'Content-Type': 'application/json', 'X-API-Token': token },
-        url: `https://api.appcenter.ms/v0.1/apps/${owner}/${app}${endpoint}`
-    };
-    return options;
-}
+async function getRulePromise(rule, context) {
+    const owner = rule.owner;
+    const app = rule.app;
+    const sourceGroup = rule.source;
+    const destinationGroup = rule.destination;
+    const maxCrashes = rule.crashes;
+    const minInstallations = rule.installs;
+    const minSessions = rule.sessions;
+
+    // We need to make a check whether the user has specified all the properties in config.
+    if (!(owner && app && sourceGroup && destinationGroup && maxCrashes && minInstallations && minSessions)) {
```

Comment:
I would split this validation part into the several `if` blocks where we check each parameter instead of checking them all at once.

Reply:
Yep will do